### PR TITLE
Mention passing errors in the docs

### LINKED ISSUE #948: Improve documentation about passing errors
Hi!

We should improve docs about when and where can/should one pass an `Error` to be correctly serialized, for example `.error(new Error("test"))"` and `.info({err: new Error("test")})` works but `.warn(new Error("test"))` does not.

Would a dev be so kind as to list out all the possible variations? So that I can submit a PR and make it clear in the docs that when does pino serialize errors and when it not.

### PR DESCRIPTION
**Pull request by vassbence**
fixes #948

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -550,6 +550,18 @@ const logger = pino(pinoOptions)
...
 
+<a id="errors"></a>
+#### Errors
+
+Errors can be supplied as either the first parameter or if already using `mergingObject` then as the `err` on the `mergingObject`.
+```js
+logger.info(new Error("test"))
+// {"level":30,"time":1531257618044,"msg":"test","stack":"...","type":"Error","pid":55956,"hostname":"x"}
+
+logger.info({err: new Error("test"), otherkey: 123}, "some text")
```
Comment (mcollina) [inline docs/api.md:561]:
```suggestion
logger.info({ err: new Error("test"), otherkey: 123 }, "some text")
```

**Review by mcollina — COMMENTED** [review]
just a nit

**Review by mcollina — APPROVED** [review]
lgtm

File: docs/api.md [hunk 2]
```diff
@@ -550,6 +550,18 @@ const logger = pino(pinoOptions)
 * See [`message` log method parameter](#message)
 * See [`logMethod` hook](#logmethod)
 
+<a id="errors"></a>
+#### Errors
+
+Errors can be supplied as either the first parameter or if already using `mergingObject` then as the `err` on the `mergingObject`.
```
Comment (jsumners) [inline docs/api.md:556]: This is accurate if the user has not customized the serializers, e.g.:

```js
const logger = pino({
  serializers: {
    error: stdSerializers.error
  }
})
```

In the above, the `err` serializer, as described in this documentation, will not be defined. Instead, it is renamed to `error`. So:

```js
// Will not serialize
logger.error({ err: Error('boom') })

// Will serialize
logger.error({ error: Error('boom') })
```

File: docs/api.md [hunk 3]
```diff
@@ -550,6 +550,18 @@ const logger = pino(pinoOptions)
 * See [`message` log method parameter](#message)
 * See [`logMethod` hook](#logmethod)
 
+<a id="errors"></a>
```
Comment (jsumners) [inline docs/api.md:553]: This identifier is very generic and will likely cause conflicts at some point. I suggest using "error-serialization" as the name.

File: docs/api.md (hunk 2 above)
Comment (vassbence) [reply docs/api.md]: Tried to point this out, let me know if you have a more concise sentence.

File: docs/api.md [hunk 4]
```diff
@@ -550,6 +550,18 @@ const logger = pino(pinoOptions)
 * See [`message` log method parameter](#message)
 * See [`logMethod` hook](#logmethod)
 
+<a id="error-serialization"></a>
+#### Errors
+
+Errors can be supplied as either the first parameter or if already using `mergingObject` then as the (by default, can be changed via the `serializers` option when creating a logger) `err` property on the `mergingObject`.
```
Comment (jsumners) [inline docs/api.md:556]:
```suggestion
Errors can be supplied as either the first parameter or if already using `mergingObject` then as the `err` property on the `mergingObject`.

> ## Note
> This section describes the default configuration. The error serializer can be
> mapped to a different key using the [`serializers`](#opt-serializers) option.
```

**Review by jsumners — APPROVED** [review]
LGTM

**Pull request merged**